fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/services/BinaryServiceImpl.java

```diff
@@ -58,9 +62,26 @@ public FedoraBinary findOrCreate(final Session session, final String path) {
 
             if (dsNode.isNew()) {
                 initializeNewDatastreamProperties(dsNode);
+
+                final Node parent = dsNode.getDepth() > 0 ? dsNode.getParent() : null;
```

Comment:
If the `dsNode` was created without a `slug` and therefore the `parent` is a PairTree, do we want to be touching the PairTree or the logical parent container? I suspect the latter.